FinTrack/Controllers/TransactionController.cs

```diff
@@ -193,4 +194,46 @@ private bool TransactionExists(int id)
     {
         return _context.Transactions.Any(e => e.Id == id);
     }
+
+    [HttpGet("filter")]
+    public async Task<ActionResult<IEnumerable<TransactionDTO>>> FilterTransactions(int userId, int? categoryId = null, DateTime? startDate = null, DateTime? endDate = null)
+    {
+        var query = _context.Transactions
+            .Include(t => t.Category)
+            .Include(t => t.User)
+            .Where(t => t.UserId == userId)
+            .AsQueryable();
+
+        if (categoryId.HasValue)
+        {
+            query = query.Where(t => t.CategoryId == categoryId.Value);
+        }
+
+        if (startDate.HasValue && endDate.HasValue)
+        {
+            query = query.Where(t => t.Date >= startDate.Value && t.Date <= endDate.Value);
+        }
+        else if (startDate.HasValue)
+        {
+            query = query.Where(t => t.Date >= startDate.Value);
+        }
+        else if (endDate.HasValue)
+        {
+            query = query.Where(t => t.Date <= endDate.Value);
+        }
+
+        var transactions = await query.ToListAsync();
```

Comment:
Ovde možeš da dobiješ malo na performance-u ako uradiš `ToListAsync()` na kraju, kada si završila sa mapiranjem. U ovom slučaju razlika verovatno nije toliko bitna, ali to je kao neki best practice, plus mislim da možeš i da skratiš kod malo jer `Select()` može da se uradi direktno nad `query`. Ovako nešto:

```
 var transactionDtos = await query.Select(t => new TransactionDTO
 {
     Amount = t.Amount,
     Date = t.Date,
     Description = t.Description,
     CategoryId = t.CategoryId,
     UserId = t.UserId,
     TransactionType = t.TransactionType
 }).ToListAsync();
```